gwvolman/lib/zenodo.py

```diff
@@ -0,0 +1,242 @@
+            stream=True,
+            jsonResp=False,
+        )
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".zip") as tmp:
+
+            # Write the zip file
+            for chunk in stream.iter_content(chunk_size=65536):
+                tmp.write(chunk)
+            tmp.seek(0)
+
+            r = self.request(
+                "/api/deposit/depositions/{}/files".format(deposition["id"]),
+                method="POST",
+                data={"filename": "{}.zip".format(self.tale["_id"])},
+                files={"file": tmp},
+            )
+
+            try:
+                r.raise_for_status()
+            except requests.HTTPError as exc:
+                msg = "Failed to upload to a deposition (id:{}).".format(
+                    deposition["id"]
+                )
+                msg += " Server returned: " + str(exc)
+                if r.status_code > 399 and r.status_code < 500:
+                    msg += "\n" + json.dumps(r.json(), sort_keys=True, indent=4) + "\n"
+                logging.warning(msg)
+                raise ValueError(msg)
+            logging.debug(
+                "[zenodo:deposit_tale] POST depositions/{}/files = {}".format(
+                    deposition["id"], r.json()
+                )
+            )
+
+    def get_tale_metadata(self):
+        """Convert the Tale metadata to a Zenodo metadata."""
+        return {
+            "metadata": {
+                "upload_type": "publication",
+                "publication_type": "other",
+                "title": self.manifest["schema:name"],
+                "creators": [
+                    {
+                        "name": "{}, {}".format(
+                            author["schema:familyName"], author["schema:givenName"]
+                        ),
+                        "orcid": author["@id"].split("/")[-1],
+                    }
+                    for author in self.manifest["schema:author"]
+                ],
+                "description": self._render_description(
+                    self.manifest["schema:description"]
+                ),
+                "access_right": "open",
+                "license": self.tale["licenseSPDX"].lower(),
+                "prereserve_doi": True,
+                "related_identifiers": [
+                    {"relation": "cites", "identifier": ds["identifier"]}
+                    for ds in self.manifest.get("Datasets", [])
+                ],
+                "references": [
+                    citation for citation in self.tale.get("dataSetCitation", [])
+                ],
+            }
+        }
+
+    def create_deposition(self):
+        """Create a deposition @ Zenodo."""
+        zenodo_metadata = self.get_tale_metadata()
+        logging.debug(
+            "[zenodo:create_deposition] metadata = {}".format(zenodo_metadata)
+        )
+
+        self.update_progress(message="Creating a deposition at " + self.resource_server)
+
+        r = self.request(
```

Comment:
We can leave this to another issue, but if I re-publish a tale, wouldn't it make sense to version the previous one instead of creating a new one (which seems to be the current behavior)? I know you're handling concept DOIs and versions on ingest.

Reply:
It's slightly problematic, cause DOI from sandbox.zenodo.org doesn't actually resolve and therefore there's no easy way to get the published deposition. I'll try to implement some work around for this...